### ISSUE TITLE
hidden-content rule does not catch <template> content

### ISSUE THREAD

**Issue by mohanraj-r** [dequelabs/axe-core#2842]
Expectation: [`hidden-content` rule](https://dequeuniversity.com/rules/axe/4.1/hidden-content?application=RuleDescription) should alert "you to the fact that there is visually hidden content on the page under test"

Actual: It does not alert about the `<template>` content

Motivation: Prevent false negatives and alert user to the fact that there are hidden/un-rendered parts in the DOM that were not analyzed. 

<pre><code>
axe-core version: 4.1.2
axe-webdriver, extension or other integration version: NA

Browser and Assistive Technology versions: NA

For Tooling issues:
- Node version: v12.21.0
- Platform:  Mac
</code></pre>

In the code below the issue with the `img` element missing alt txt is not reported.

```js
		const template = document.createElement('template');
		template.innerHTML = `<img src="foo.jpg" />`;
		document.body.appendChild(template);
		const results = await axe.run(template, {rules: {'hidden-content': {enabled: true}}});
		console.log(results.violations); // Output: "[]"
```
 
But if the `img` is created inside a normal container (e.g. `div`) it works as expected.
Technically `template` is not visually hidden but just not rendered by definition. But it would be great to get this covered as well to prevent false negatives.

**Comment by dylanb** [comment]
`<template>` content is not hidden content of the sort intended by this rule, There is also no way for us to determine whether the template has been used AFAIK so this would also generate lots of false positives which goes against our manifesto
